gaia-toolbar.js

```diff
@@ -1,278 +1,354 @@
 
 /**
- * Locals
+ * Exports
  */
 
-var forEach = [].forEach;
-var slice = [].slice;
+module.exports = component.register('gaia-toolbar', {
+  created: function() {
+    this.setupShadowRoot();
 
-/**
- * Extend from the `HTMLElement` prototype
- *
- * @type {Object}
- */
-var proto = Object.create(HTMLElement.prototype);
+    this.els = {
+      inner: this.shadowRoot.querySelector('.inner'),
+      moreButton: this.shadowRoot.querySelector('.more-button')
+    };
 
-proto.createdCallback = function() {
-  this.createShadowRoot().innerHTML = template;
+    // Setup initial attributes
+    this.l10nMoreId = this.getAttribute('l10n-more-id');
+    this.l10nMoreValue = this.getAttribute('l10n-more-value');
 
-  this.els = {
-    inner: this.shadowRoot.querySelector('.inner'),
-    moreButton: this.shadowRoot.querySelector('.more-button')
-  };
+    this.els.moreButton.addEventListener('click', this.openOverflow.bind(this));
 
-  this.els.moreButton.addEventListener('click', this.openOverflow.bind(this));
+    // process everything that doesn't affect user interaction
+    // after the component is created
+    setTimeout(() => {
+      this.makeAccessible();
+      this.localize();
+    });
 
-  var self = this;
-  this.reflow_raf = rafWrap(this.reflow, this);
-  addEventListener('resize', this.reflow_raf);
-  this.shadowStyleHack();
+    this.reflow_raf = rafWrap(this.reflow, this);
+    addEventListener('resize', this.reflow_raf);
+    this.shadowStyleHack();
 
-  if (document.readyState === 'loading') {
-    addEventListener('load', this.reflow.bind(this));
-    return;
-  }
+    if (document.readyState === 'loading') {
+      addEventListener('load', this.reflow.bind(this));
+      return;
+    }
+
+    this.style.visibility = 'hidden';
+    this.reflow_raf();
+  },
 
-  this.style.visibility = 'hidden';
-  self.reflow_raf();
-};
-
-proto.attachedCallback = function() {
-  this.style.visibility = 'hidden';
-  this.reflow_raf();
-};
-
-proto.reflow = function() {
-  this.release();
-  var overflow = this.getOverflowData();
-  if (overflow.overflowed) { this.onOverflowed(overflow); }
-  this.style.visibility = '';
-};
-
-proto.getOverflowData = function() {
-  var space = this.els.inner.clientWidth;
-  var child = this.children[0];
-  var overflowed = false;
-  var willFit = 0;
-  var total = 0;
-  var overflow;
-  var width;
-
-  while (child) {
-    width = child.clientWidth;
-    overflow = width + total - space;
-
-    if (overflow > 3) {
-      overflowed = true;
-      break;
+  makeAccessible: function() {
+    this.setAttribute('role', 'toolbar');
+
+    // If there's more than one toolbar present, toolbar must have a label.
+    if (document.getElementsByTagName('gaia-toolbar').length > 1 &&
+      !this.getAttribute('data-l10n-id') && !this.getAttribute('aria-label')) {
+        console.error('Component not fully accessible: please provide a label' +
+          'for the toolbar itself.', this);
```

Comment:
So I'm still not sure about this one. My personal preference is that it would console as error for a couple of reasons:
* A toolbar without a label (with multiple toolbars on the page) is not a valid based on the spec: http://www.w3.org/TR/wai-aria/roles#toolbar
* It is a bigger problem than a place were warning is invoked: https://github.com/gaia-components/gaia-toolbar/pull/2/files#diff-6131b2d50a8cf7654d3e0597d37f00f9R205
* Contextually this case really hampers the UX for the screen reader user. There is an assumption that if a toolbar has no name it is unique to the page. Though I have not seen our stock apps with multiple toolbars, if that ever happens, the user will automatically assume that they are somehow back to the toolbar they have seen before.